cpp/models/abm/infection.cpp

```diff
@@ -0,0 +1,143 @@
+/* 
+* Copyright (C) 2020-2021 German Aerospace Center (DLR-SC)
+*
+* Authors: David Kerkmann, Sascha Korf
+*
+* Contact: Martin J. Kuehn <Martin.Kuehn@DLR.de>
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "abm/infection.h"
+#include <math.h>
+
+namespace mio
+{
+namespace abm
+{
+
+ViralLoad::ViralLoad(const TimePoint& start_date, const GlobalInfectionParameters& params)
+    : m_start_date(start_date)
+{
+    draw_viral_load(params);
+}
+
+void ViralLoad::draw_viral_load(const GlobalInfectionParameters& /*params*/)
+{
+    // These numbers are subject to change, They are going to be based on distributions backed from data.
+    m_peak     = 5.0;
+    m_incline  = 1.0;
+    m_decline  = -1.0;
+    m_end_date = m_start_date + TimeSpan(int(m_peak / m_incline - m_peak / m_decline));
+}
+
+double ViralLoad::get_viral_load(const TimePoint& t) const
+{
+    if (t >= m_start_date && t <= m_end_date) {
+        if (t.seconds() <= m_start_date.seconds() + m_peak / m_incline) {
+            return m_incline * (t - m_start_date).seconds();
+        }
+        else {
+            return m_peak + m_decline * (t.seconds() - m_peak / m_incline - m_start_date.seconds());
+        }
+    }
+    else {
+        return 0.;
+    }
+}
+
+Infection::Infection(const VirusVariant& virus, const GlobalInfectionParameters& params, const TimePoint& start_date,
+                     const InfectionState& start_state, const bool detected)
+    : m_virus_variant(virus)
+    , m_viral_load(start_date, params)
+    , m_detected(detected)
+{
+    draw_infection_course(start_date, params, start_state);
+};
+
+double Infection::get_viral_load(const TimePoint& t) const
+{
+    return m_viral_load.get_viral_load(t);
+}
+
+double Infection::get_infectivity(const TimePoint& t) const
+{
+    return 1 / (1 + exp(-(m_log_norm_alpha + m_log_norm_beta * get_viral_load(t))));
```

Comment:
I removed the get_viral_load function from the Infection, but in general, as the code might change, I think it is useful to have both layers to also use the viral load for symptoms, for example, in case there is a connection.